DB/api/detect_events.py

```diff
@@ -0,0 +1,112 @@
+from itertools import cycle
+
+
+def detect(time_series, threshold):
+	"""
+	take in a series of time and power readings and return a list of events
+
+	:param time_series: a generator created by the retrieve_within_filters function
+						contains dates and power readings
+	:param threshold: the threshold value for event detection
+	:return eventList: a python matrix containing a list of times for detected events
+	"""
+	if threshold == 0:
+		raise Exception("Threshold cannot be 0, try another value.")
+
+	event_list = []
+
+	t1 = time_series[1]
+	t2 = time_series[2]
+	t3 = time_series[3]
+	t4 = time_series[4]
+	t5 = time_series[5]
+	t6 = time_series[6]
+
+	prev_avg = (float(t1[1]) + float(t2[1]) + float(t3[1]) + float(t4[1]) + float(t5[1])) / 5
+	curr_avg = (float(t2[1]) + float(t3[1]) + float(t4[1]) + float(t5[1]) + float(t6[1])) / 5
```

Comment:
Oh, well it would be better to use a map first so you only have to do it once. In Python 3, maps return a generator.